XenModel/Utils/Helpers.cs

```diff
@@ -1537,5 +1539,46 @@ public static bool ConnectionRequiresRbac(IXenConnection connection)
 
             return GetCoordinator(connection).external_auth_type != Auth.AUTH_TYPE_NONE;
         }
+
+        /// <summary>
+        /// Adds the specified authentication token to the existing query string and returns
+        /// the modified query string.
+        /// </summary>
+        /// <param name="authToken">The authentication token to add to the query string.</param>
+        /// <param name="existingQueryString">The existing query string to add the token to.</param>
+        /// <returns>The modified query string.</returns>
+        public static string AddAuthTokenToQueryString(string authToken, string existingQueryString)

```

Comment:
While this works it is a little overkill, we have a preformatted auth token that you pull apart and then put back together.

Reply:
I could add it with simple string concatenation but we wouldn't have any sanity check mechanism to go with it. You could argue it's fine since we know right now what's being used etc, but it's not really future proofed. I went for future proofing this time and so I sticked with suggestions in the Microsoft docs

Reply:
So basically I agree with you, but I usually lean towards adding a bit more checks to avoid potential issues in the future, even if unlikely and with low impact, like in this case